test/bin/scenario.sh

```diff
@@ -0,0 +1,378 @@
+    # hypervisor forwarding connections. If we have no PUBLIC_IP, use
+    # the VM IP and assume a local connection.
+    if [ -n "${PUBLIC_IP}" ]; then
+        echo "${PUBLIC_IP}" > "${SCENARIO_INFO_DIR}/${SCENARIO}/vms/${vmname}/public_ip"
+    else
+        echo "${ip}" > "${SCENARIO_INFO_DIR}/${SCENARIO}/vms/${vmname}/public_ip"
+    fi
+
+    enable_ip_access "${full_vmname}" "${ip}"
+    wait_for_greenboot "${full_vmname}" "${ip}"
+
+    echo "${full_vmname} is up and ready"
+}
+
+# Clean up the resources for one VM.
+remove_vm() {
+    local vmname="${1}"
+
+    local full_vmname
+    full_vmname="$(full_vm_name "${vmname}")"
+
+    # Remove the actual VM and its storage
+    if sudo virsh dumpxml "${full_vmname}" >/dev/null; then
+        sudo virsh destroy "${full_vmname}"
+        sudo virsh undefine "${full_vmname}"
+    fi
+    if sudo virsh vol-dumpxml "${full_vmname}.qcow2" vm-storage >/dev/null; then
+        sudo virsh vol-delete "${full_vmname}.qcow2" vm-storage
+    fi
+
+    # Remove the info file so something processing the VMs does not
+    # assume the file exists. This is most useful in a local setting.
+    rm -rf "${SCENARIO_INFO_DIR}/${SCENARIO}/vms/${vmname}"
+}
+
+# Run the tests for the current scenario
+run_tests() {
+    local vm_name="${1}"
+    shift
+
+    echo "Running tests with $# args" "$@"
+
+    if [ ! -d "${RF_VENV}" ]; then
+        error "RF_VENV (${RF_VENV}) does not exist, create it with: ${ROOTDIR}/scripts/fetch_tools.sh robotframework"
+        exit 1
+    fi
+    local rf_binary="${RF_VENV}/bin/robot"
+    if [ ! -f "${rf_binary}" ]; then
+        error "robot is not installed to ${rf_binary}"
+        exit 1
+    fi
+
+
+    local ssh_port_file="${SCENARIO_INFO_DIR}/${SCENARIO}/vms/${vm_name}/ssh_port"
+    local api_port_file="${SCENARIO_INFO_DIR}/${SCENARIO}/vms/${vm_name}/api_port"
+    local lb_port_file="${SCENARIO_INFO_DIR}/${SCENARIO}/vms/${vm_name}/lb_port"
+    local public_ip_file="${SCENARIO_INFO_DIR}/${SCENARIO}/vms/${vm_name}/public_ip"
+    local ip_file="${SCENARIO_INFO_DIR}/${SCENARIO}/vms/${vm_name}/ip"
+    local api_port
+    local ssh_port
+    local public_ip
+    local vm_ip
+    local f
+    for f in "${ssh_port_file}" "${api_port_file}" "${lb_port_file}" "${public_ip_file}" "${ip_file}"; do
+        if [ ! -f "${f}" ]; then
+            error "Cannot read ${f}"
+            exit 1
+        fi
+    done
+    ssh_port=$(cat "${ssh_port_file}")
+    api_port=$(cat "${api_port_file}")
+    lb_port=$(cat "${lb_port_file}")
+    public_ip=$(cat "${public_ip_file}")
+    vm_ip=$(cat "${ip_file}")
+
+    local variable_file="${ARTIFACT_DIR}/e2e/${SCENARIO}/variables.yaml"
+    echo "Writing variables to ${variable_file}"
+    mkdir -p "$(dirname "${variable_file}")"
+    cat - <<EOF | tee "${variable_file}"
+VM_IP: ${vm_ip}
+API_PORT: ${api_port}
+LB_PORT: ${lb_port}
+USHIFT_HOST: ${public_ip}
+USHIFT_USER: redhat
+SSH_PRIV_KEY: ""
```

Comment:
I guess this need to be configurable?

Reply:
Yes, I have a note to do that as part of the next step of integrating these scripts with the CI system, where we have a key but no agent.

Reply:
Done